runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,901 @@
+        )["logGroups"]
+        lambdas_log_groups = [group["logGroupName"] for group in lambdas_log_groups]
+        return lambdas_log_groups
+
+    @contextlib.contextmanager
+    def _wait_till_log_group_is_created(
+        self, curr_lambda_log_group, lambdas_log_groups
+    ):
+        time_passed = 0
+        while curr_lambda_log_group not in lambdas_log_groups:
+            if time_passed > self.MAX_WAIT_TIME:
+                raise TimeoutError(
+                    f"The log group called {curr_lambda_log_group} is being deployed in AWS cloudwatch for too long. "
+                    + " Please check the resource in AWS console, delete relevant resource(s) "
+                    + "if necessary, and re-run your Runhouse code."
+                )
+            time.sleep(1)
+            lambdas_log_groups = self._get_log_group_names()
+            time_passed += 1
+        yield True
+
+    def _invoke(self, *args, **kwargs) -> Any:
+        lambdas_log_groups = self._get_log_group_names()
+        curr_lambda_log_group = f"{LOG_GROUP_PREFIX}{self.name}"
+        invoke_for_the_first_time = curr_lambda_log_group not in lambdas_log_groups
+
+        payload_invoke = {}
+
+        if len(args) > 0 and self.args_names is not None:
+            payload_invoke = {self.args_names[i]: args[i] for i in range(len(args))}
+        invoke_res = self.lambda_client.invoke(
+            FunctionName=self.name,
+            Payload=json.dumps({**payload_invoke, **kwargs}),
+            LogType="Tail",
+        )
+        return_value = invoke_res["Payload"].read().decode("utf-8")
+
+        try:
+            logger.error(invoke_res["FunctionError"])
+            return_value = json.loads(return_value)
+            raise RuntimeError(
+                f"Failed to run {self.name}. Error: {return_value['errorType']}, {return_value['errorMessage']}"
+            )
+
+        except KeyError:
+            if invoke_for_the_first_time:
+                lambdas_log_groups = self._get_log_group_names()
+                with self._wait_till_log_group_is_created(
+                    curr_lambda_log_group, lambdas_log_groups
+                ):
+                    self.logs_client.put_retention_policy(
+                        logGroupName=curr_lambda_log_group,
+                        retentionInDays=self.retention_time,
+                    )
+
+            log_lines = "Function Logs are:\n" + base64.b64decode(
+                invoke_res["LogResult"]
+            ).decode("utf-8")
+
+            print(log_lines)
+            return return_value
+
+    def map(self, *args, **kwargs):
+        """Map a function over a list of arguments.
+
+        Example:
+            >>> # The my_lambda_handler.py file
+            >>> def my_summer(arg1, arg2, arg3):
+            >>>     return arg1 + arg2 + arg3
+            >>>
+            >>> # your 'main' python file, where you are using runhouse
+            >>> summer_lambda = rh.lambda_function(
```

Comment:
similar to my other comment, but the empty `.to()` call is a little confusing. Aren't we calling it automatically anyway (unless its a dryrun)? Instead maybe we can highlight the usage path of calling `.to("aws_lambda")`?